cache-prefetch.go

```diff
@@ -0,0 +1,183 @@
+package rdns
+
+import (
+	"errors"
+	"expvar"
+	"github.com/miekg/dns"
+	"github.com/sirupsen/logrus"
+	"sync"
+	"time"
+)
+
+type CachePrefetch struct {
+	CachePrefetchOptions
+	id       string
+	resolver Resolver
+	mu       sync.Mutex
+	metrics  *CachePrefetchMetrics
+}
+
+
+type CachePrefetchMetrics struct {
+	// Cache hit count.
+	domainEntries map[string]CachePrefetchEntry
+}
+type CachePrefetchEntry struct {
+	// request hit count.
+	hit 			  expvar.Int
```

Comment:
Probably best to stick to the base types, and make the whole entry thread-safe.

Reply:
Are you saying `int32` instead of `expvar.Int`??